### ISSUE TITLE
Matplotlib not working

### ISSUE THREAD

**Issue by tacheraSasi** [pyscript/pyscript#2267]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

I tried  running a simple python code with matplotlib and didnot work 

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>PyScript Currency Converter</title>
    <script defer src="https://pyscript.net/alpha/pyscript.js"></script>
    <link rel="stylesheet" href="https://pyscript.net/alpha/pyscript.css">
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 20px;
        }
        .converter-container {
            margin-bottom: 20px;
        }
        .output {
            font-weight: bold;
            color: green;
        }
    </style>
</head>
<body>

    <h1>Currency Converter: USD to EUR</h1>
    <div class="converter-container">
        <label for="usd">Enter amount in USD:</label>
        <input type="number" id="usd" placeholder="Amount in USD">
        <button id="convert-btn">Convert</button>
    </div>

    <div class="output">
        <p>Converted amount in EUR: <span id="eur-result">--</span></p>
    </div>

    <h2>Historical Conversion Rates</h2>
    <div>
        <py-script>
            import matplotlib.pyplot as plt
            import numpy as np

            # Simulated historical rates for USD to EUR
            days = np.arange(1, 11)
            rates = [0.90, 0.92, 0.91, 0.93, 0.89, 0.88, 0.87, 0.91, 0.92, 0.90]

            plt.plot(days, rates, marker='o')
            plt.title("USD to EUR Rates (Past 10 Days)")
            plt.xlabel("Days")
            plt.ylabel("Exchange Rate")
            plt.grid(True)
            plt.show()
        </py-script>
    </div>

    <py-script>
        # Function to convert USD to EUR
        def convert_usd_to_eur(usd_amount, exchange_rate=0.92):
            return round(usd_amount * exchange_rate, 2)

        # Handling button click
        def on_convert_click(evt):
            usd_amount = float(Element("usd").element.value)
            eur_amount = convert_usd_to_eur(usd_amount)
            Element("eur-result").element.innerText = f"{eur_amount} EUR"

        # Attach the button click event to the Python function
        Element("convert-btn").element.onclick = on_convert_click
    </py-script>

</body>
</html>
```

**Comment by WebReflection** [comment]
`https://pyscript.net/alpha/pyscript.js` ??? ... where does that come from? 🤔 /cc @ntoll

**Comment by WebReflection** [comment]
@tacheraSasi try this one: https://pyscript.com/@examples/matplotlib/latest?files=README.md

**Comment by tacheraSasi** [comment]
Okay

**Comment by WebReflection** [comment]
@tacheraSasi happy to help but ... seriously, where did you get that URL? we just recently sunset `latest` ... there are people using `alpha` from I don't even know how long ago? If that's one of our articles, we should act to avoid further confusion about our project ... it hurts new-comers as well as ourselves, so I'd love to learn where or why people would try that link at all to grab this project, thanks!

**Comment by ntoll** [comment]
Yeah... so old and out-of-date docs are things we can't fix. But I've put a request in to our infra team to delete the `alpha` endpoint (it'll just return a 404 with a message saying to use a pinned version of PyScript).

**Comment by JelleAalbers** [comment]
> But I've put a request in to our infra team to delete the `alpha` endpoint (it'll just return a 404 with a message saying to use a pinned version of PyScript).

Would it be possible to re-upload the alpha code somewhere? Or perhaps let us know which specific snapshot / commit it refers to?

My colleague @FrankNicoMol and I run a python course that uses pyscript alpha in the student's exam environment. We would love to switch to a more recent and pinned version, but it will take us some time to figure out how to convert our content. Thank you for your work keeping pyscript alive!

**Comment by ntoll** [comment]
Hi @JelleAalbers - I can't be certain what version of PyScript was deployed at the `/alpha` endpoint. However, it would have been something very close to this release: https://github.com/pyscript/pyscript/releases/tag/2023.05.1 (details here: https://pyscript.net/releases/2023.05.1/) Please do let us know how you get on... but also update to a modern pinned version too..! 🤗 😉

**Comment by Neon22** [comment]
@JelleAalbers @FrankNicoMol I offer to help you convert the students code to a more modern version.
send me email on mschafer@wireframe.biz

**Comment by JelleAalbers** [comment]
Thanks for the kind help and offers of help! The current course (run by my colleague) is switching to a different solution, but I will teach again in fall and will try to move to a new version.

In case it is useful to anyone: we are/were using py-repl, which was removed in 2023.11.01. I've briefly tried (m)py-editor with newer versions, but the docs clearly warn it's work in progress (and I indeed couldn't get the examples to work).

The 2023.5.1 release got the py-repl to work with some changes (see the gist [here](https://gist.github.com/JelleAalbers/3e73dee7b1db1694abc479fa80c7c7e1)). Unfortunately error handling doesn't seem to work the same way anymore. If the user enters code that produces an exception, there is only a message in the browser console and nothing changes on the page. (Alpha showed an imperfectly formatted but still useful red traceback.)

**Comment by Neon22** [comment]
Also consider trying the editor on the ltk. It uses codemirror. I can't speak for the error handling.

**Comment by WebReflection** [comment]
@JelleAalbers we have a MicroPython repl on [pyscript.net](https://pyscript.net/) as first thing we showcase ... it's as simple as this:

```html
<script type="py" terminal worker>
    import code
    code.interact()
</script>
```

for pyodide, but it can work on the main and MicroPython too via:

```html
<script type="mpy" terminal>
    import code
    code.interact()
</script>
```

Anyway, closing this as we don't support *alpha* versions from years ago but happy to help or expand further, if needed.
